python/ray/util/queue.py

```diff
@@ -87,22 +130,76 @@ def get(self, block=True, timeout=None):
             else:
                 return ray.get(self.actor.get.remote(timeout))
 
+    async def get_async(self, block=True, timeout=None):
+        """Gets an item from the queue.
+
+        There is no guarantee of order if multiple consumers get from the
+        same empty queue.
+
+        Returns:
+            The next item in the queue.
+        Raises:
+            Empty: if the queue is empty and blocking is False.
+            Empty: if the queue is empty, blocking is True, and it timed out.
+            ValueError: if timeout is negative.
+        """
+        if not block:
+            try:
+                return await self.actor.get_nowait.remote()
+            except asyncio.QueueEmpty:
+                raise Empty
+        else:
+            if timeout is not None and timeout < 0:
+                raise ValueError("'timeout' must be a non-negative number")
+            else:
+                return await self.actor.get.remote(timeout)
+
     def put_nowait(self, item):
         """Equivalent to put(item, block=False).
 
         Raises:
-            Full if the queue is full.
+            Full: if the queue is full.
         """
         return self.put(item, block=False)
 
+    def put_nowait_batch(self, items):
+        """Takes in a list of items and puts them into the queue in order.
+
+        Raises:
+            Full: if the items will not fit in the queue
+        """
+        if not isinstance(items, list):
+            raise TypeError("Argument 'items' must be a list")
```

Comment:
should we support any iterable here?

Reply:
The only concrete downside I can think of is if you inadvertently call put_batch("my_string") instead of put_batch(["my_string"]), it will put each individual letter into the queue.  

The upside is that if the user has many items stored in a set, tuple or other non-list object, they'll save on runtime by not having to convert it to a list first.  So I think it's worth it to support any iterable, unless there's some other downside I'm overlooking.